target/linux/mediatek/patches-6.6/902-arm-add-cmdline-replace.patch

```diff
@@ -0,0 +1,100 @@
+--- a/arch/arm64/Kconfig
++++ b/arch/arm64/Kconfig
+@@ -2316,6 +2316,15 @@ config CMDLINE_OVERRIDE
+ 	  devices. This setting enables using "chosen/cmdline-override" as the
+ 	  cmdline if it exists in the device tree.
+ 
++config CMDLINE_REPLACE
++       bool "Replace specific cmdline from bootloader"
++	default n
++       help
++         Some bootloaders pass specific bootargs to the linux kernel that need
++         to additional behaviors for proper operation. This setting enables
++         replacing bootargs from "chosen/cmdline-replace" at early boot if it
++         exists in the device tree.
++
+ config EFI_STUB
+ 	bool
+ 
+--- a/drivers/of/fdt.c
++++ b/drivers/of/fdt.c
+@@ -1150,6 +1150,29 @@ int __init early_init_dt_scan_memory(voi
+ 	return found_memory;
+ }
+ 
++#ifdef CONFIG_CMDLINE_REPLACE
++static char *string_replace(char *str, const char *old, const char *new, size_t strlenmax) {
++	char *p = strstr(str, old);
++	size_t olen, nlen, rlen;
++
++	if (p == NULL)
++		return str;
++
++	olen = strlen(old);
++	nlen = strlen(new);
++	rlen = strlenmax - (p - str);
++
++	if (rlen < nlen)
++		return str;
```

Comment:
I think the calculation of `rlen` is wrong.

Reply:
ah yes, I feel the same thing, need to fixed

Reply:
fixed calculation of length checks and test verified with a device boot.